src/site/content/en/react/quicklink-react-spa/index.md

```diff
@@ -0,0 +1,133 @@
+    },
+    {
+      type: 'script',
+      href: '/static/js/blog.35421503.chunk.js',
+    },
+  ],
+}
+```
+
+This manifest file can be requested in two ways:
+
+- By URL, e.g. `https://site_url/rmanifest.json`.
+- Through the window object, at `window.__rmanifest`.
+
+## Prefetch chunks for in-viewport routes
+
+Once the manifest file is available, the next step is to install `quicklink` by running `npm install quicklink`.
+
+Then, the [higher order component (HOC)](https://reactjs.org/docs/higher-order-components.html) `withQuicklink()` can be used to indicate that a given route should be prefetched when the link comes into the view.
+
+The following code belongs to an `App` component of a React app that renders a top menu with four links:
+
+```jsx
+const App = () => (
+  <div className={style.app}>
+    <Hero />
+    <main className={style.wrapper}>
+      <Suspense fallback={<div>Loading...</div>}>
+        <Route path="/" exact component={Home} />
+        <Route path="/blog" exact component={Blog} />
+        <Route path="/blog/:title" component={Article} />
+        <Route path="/about" exact component={About} />
+      </Suspense>
+    </main>
+    <Footer />
+  </div>
+);
+```
+
+To tell `quicklink` that these routes should be prefetched as they come into the view:
+
+1. Import the `quicklink` HOC at the beginning of the component.
+1. Wrap each route with the `withQuicklink()` HOC, passing the page component and options parameter to it.
+
+```html
+const options = {
+  origins: [],
+};
+const App = () => (
+  <div className={style.app}>
+    <Hero />
+    <main className={style.wrapper}>
+      <Suspense fallback={<div>Loading...</div>}>
+        <Route path="/" exact component={withQuicklink(Home, options)} />
+        <Route path="/blog" exact component={withQuicklink(Blog, options)} />
+        <Route
+          path="/blog/:title"
+          component={withQuicklink(Article, options)}
+        />
+        <Route path="/about" exact component={withQuicklink(About, options)} />
+      </Suspense>
+    </main>
+    <Footer />
+  </div>
+);
+```
+
+The `withQuicklink()` HOC uses the path of the route as a key to obtain its associated chunks from `rmanifest.json`.
+Under the hood, as links come into the view, the library injects a `<link rel="prefetch">` tag in the page for each chunk so they can be prefetched.
+Prefetched resources will be requested at the lowest priority by the browser and kept in the [HTTP Cache](https://web.dev/http-cache/) for 5 minutes, after which point, the `cache-control` rules of the resource apply.
+As a result of this, when a user clicks on a link and moves to a given route, the chunks will be retrieved from the cache, greatly improving the time it takes to render that route.
+
+## Conclusion
+
+Prefetching can greatly improve load times for future navigations. In React single page apps, this can be achieved by loading the chunks associated with each route, before the user lands in them.
```

Comment:
```suggestion
Prefetching can greatly improve load times for future navigations. In React single-page apps, this can be achieved by loading the chunks associated with each route, before the user lands in them.
```